src/trans-word-level/instantiate_word_level.cpp

```diff
@@ -175,24 +169,131 @@ wl_instantiatet::instantiate_rec(exprt expr, const mp_integer &t) const
+    expr.id() == ID_sva_non_overlapped_implication)
+  {
+    auto &implication = to_binary_expr(expr);
+    std::vector<std::pair<mp_integer, exprt>> result;
+
+    // This is the product of the match points on the LHS and RHS
+    const auto lhs_match_points = instantiate_sequence(implication.lhs(), t);
+    for(auto &lhs_match_point : lhs_match_points)
+    {
+      // The RHS of the non-overlapped implication starts one timeframe later
+      auto t_rhs = expr.id() == ID_sva_non_overlapped_implication
+                     ? lhs_match_point.first + 1
+                     : lhs_match_point.first;
+
+      // Do we exceed the bound? Make it 'true'
+      if(t_rhs >= no_timeframes)
+        return {{no_timeframes - 1, true_exprt()}};
+
+      const auto rhs_match_points =
+        instantiate_sequence(implication.rhs(), t_rhs);
+
+      for(auto &rhs_match_point : rhs_match_points)
+      {
+        exprt cond;
+        if(expr.id() == ID_sva_sequence_concatenation)
+        {
+          cond = and_exprt{lhs_match_point.second, rhs_match_point.second};
+        }
+        else if(
+          expr.id() == ID_sva_overlapped_implication ||
+          expr.id() == ID_sva_non_overlapped_implication)
+        {
+          cond = implies_exprt{lhs_match_point.second, rhs_match_point.second};
+        }
+        else
+          PRECONDITION(false);
+
+        result.push_back({rhs_match_point.first, cond});
+      }
+    }
+
+    return result;
+  }
+  else if(expr.id() == ID_sva_sequence_intersect)
+  {
+    PRECONDITION(false);
+  }
+  else if(expr.id() == ID_sva_sequence_first_match)
+  {
+    PRECONDITION(false);
+  }
+  else if(expr.id() == ID_sva_sequence_throughout)
+  {
+    PRECONDITION(false);
+  }
+  else if(expr.id() == ID_sva_sequence_within)
+  {
+    PRECONDITION(false);
+  }
+  else
   {
-    // much like regular 'and'
-    expr.id(ID_and);
-    return instantiate_rec(expr, t);
+    // not a sequence, evaluate as state predicate
+    return {instantiate_rec(expr, t)};
+  }
+}
+
+/*******************************************************************\
+
+Function: wl_instantiatet::instantiate_rec
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+std::pair<mp_integer, exprt>
+wl_instantiatet::instantiate_rec(exprt expr, const mp_integer &t) const
+{
+  expr.type() = instantiate_rec(expr.type(), t);
+
+  if(expr.id() == ID_next_symbol)
+  {
+    expr.id(ID_symbol);
+    auto u = t + 1;
+    return {u, timeframe_symbol(u, to_symbol_expr(std::move(expr)))};
+  }
+  else if(expr.id() == ID_symbol)
+  {
+    return {t, timeframe_symbol(t, to_symbol_expr(std::move(expr)))};
+  }
+  else if(is_SVA_sequence(expr))
+  {
+    // sequence expressions -- these may have multiple potential
+    // match points, and evaluate to true if any of them matches
+    const auto match_points = instantiate_sequence(expr, t);
+    exprt::operandst disjuncts;
+    mp_integer max = t;
```

Comment:
`t` isn't necessarily one of the values of `match_point.first`, is that a problem?

Reply:
All the `match_point.first` are `>=t`, and hence, `max` is going to be one of them unless there is no match point at all.